route_verification/bgp/src/cmp/filter.rs

```diff
@@ -91,13 +91,25 @@ impl<'a> CheckFilter<'a> {
 
     /// Check for this case:
     /// - The AS number itself is the `<filter>`.
-    /// - Exporting customers routes.
+    /// - Exporting routes received from customers.
     #[inline]
     pub fn is_filter_export_customer(&self, num: u32, op: RangeOperator) -> bool {
         self.export
             && self.cmp.verbosity.check_customer
             && num == self.self_num
-            && self.filter_as_set(&customer_set(num), op).is_none()
+            && op.permits(&self.cmp.prefix)
+            && self.is_prev_as_customer()
+    }
+
+    /// Check if the previous ASN on the AS path is in our customer set.
+    fn is_prev_as_customer(&self) -> bool {
+        match self.last_on_path() {
```

Comment:
Does this return the origin AS or the previous AS (relative to the current AS) on the path? I think we want the previous AS on the path here.

Reply:
This returns the previous AS on the AS path relative to the AS under check (the exporting one, in this case).